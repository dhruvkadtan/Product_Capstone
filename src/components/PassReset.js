import { Field, Form, Formik } from "formik";
import React from "react";

import productSchema from "../data/productSchema";
import {toast, ToastContainer} from 'react-toastify';
import "../../node_modules/react-toastify/dist/ReactToastify.css"
import { useHistory } from "react-router-dom";

const PassReset = () =>{

    let history = useHistory();

    const handleReset = () => {
        toast.info("Reset Link sent to email successfully", {position: toast.POSITION.TOP_CENTER,autoClose:2000});
        setTimeout(() => document.getElementById("back").click(),3000)
    }
        return (
            <div className="container-xxl">
                <div className="container">
                    <div className="text-center mx-auto mb-5 mt-3">
                        <h3 className="text-secondary">Forgot Password</h3>
                    </div>
                    <div  className="row justify-content-center">
                        <div className="col-lg-6 ">
                            <Formik initialValues={{userId : ""}}
                                    validationSchema={productSchema}
                                    
                            >
                                {  ({touched,errors}) => (
                                    <Form>
                                        <div className="row ms-5 me-5">
                                            <div className="col-12 mb-3">               
                                                <label className="form-label text-secondary fw-bold" htmlFor="userId">Enter your email: </label>
                                                    <Field
                                                        type="email"
                                                        name="userId"
                                                        placeholder="Enter your user email"
                                                        className="form-control"
                                                                />
                                                    {touched.userId && errors.userId && <span style={{color : 'red'}}>{errors.userId}</span>}                   
                                             </div>
                                            <div className="col-md-6 text-center mb-3">
                                                <button className="form-control btn btn-secondary rounded-pill fw-bold"
                                                        onClick={handleReset}
                                                        type="button">Reset</button>
                                                <ToastContainer/>
                                            </div>
                                            <div className="col-md-6 text-center mb-3">
                                                <button  className="form-control btn btn-secondary rounded-pill fw-bold"
                                                            onClick={() => history.goBack()}>
                                                    Back
                                                </button>
                                            </div>
                                        </div>
                                    </Form>
                                )}
                            </Formik>
                        </div>
                    </div>
                </div>   
            </div>
        )
        
}
export default PassReset;

/*   <NavLink className="form-control btn btn-secondary rounded-pill fw-bold"
                                                to="/signin">Back</NavLink> */