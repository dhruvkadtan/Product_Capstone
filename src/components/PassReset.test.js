import React from 'react';
import renderer from 'react-test-renderer';
import  PassReset from './PassReset';
import { screen,render as renderRtl } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const setup = () => renderRtl(<PassReset/>)

describe('checking render',() => {
  test('render correctly', () => {
    const tree = renderer.create(<PassReset/>)
    expect(tree).toMatchSnapshot();
  })

})

describe("checking dom elements", () => {
  
  test('Checking heading', () => {
    setup();
    const heading = screen.getByText("Forgot Password")
    expect(heading).toBeInTheDocument();
  })

  test('checking reset button', () => {
    setup();
    const resButton = screen.getByRole('button', {name: 'Reset'});
        expect(resButton).toBeVisible();
  })

  test('checking back button', () => {
    setup();
    const backButton = screen.getByRole('button', {name: 'Back'});
        expect(backButton).toBeVisible();
  })

  
})
