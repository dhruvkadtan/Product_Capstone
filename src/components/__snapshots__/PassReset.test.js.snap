// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking render render correctly 1`] = `
<div
  className="container-xxl"
>
  <div
    className="container"
  >
    <div
      className="text-center mx-auto mb-5 mt-3"
    >
      <h3
        className="text-secondary"
      >
        Forgot Password
      </h3>
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="col-lg-6 "
      >
        <form
          action="#"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="row ms-5 me-5"
          >
            <div
              className="col-12 mb-3"
            >
              <label
                className="form-label text-secondary fw-bold"
                htmlFor="userId"
              >
                Enter your email: 
              </label>
              <input
                className="form-control"
                name="userId"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your user email"
                type="email"
                value=""
              />
            </div>
            <div
              className="col-md-6 text-center mb-3"
            >
              <button
                className="form-control btn btn-secondary rounded-pill fw-bold"
                onClick={[Function]}
                type="button"
              >
                Reset
              </button>
              <div
                className="Toastify"
              />
            </div>
            <div
              className="col-md-6 text-center mb-3"
            >
              <button
                className="form-control btn btn-secondary rounded-pill fw-bold"
                onClick={[Function]}
              >
                Back
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
